### PR TITLE
removed default credentials from the bitbucket server config form

### PR DESCRIPTION
**Pull request by alibk0rd**
\+ related backend and test changes
(Related to [JENKINS-62235](https://issues.jenkins-ci.org/browse/JENKINS-62235))

### CODE REVIEW

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java [hunk 1]
```diff
@@ -359,27 +321,20 @@ public FormValidation doTestConnection(
...
+                    return FormValidation.error(
+                            "We can't connect to Bitbucket Server. Choose a different personal access token with project admin permissions");
                 }
-                BitbucketClientFactory client =
-                        clientFactoryProvider.getClient(
-                                config.getBaseUrl(),
-                                jenkinsToBitbucketCredentials.toBitbucketCredentials(credentials.orElse(null), config.getGlobalCredentialsProvider(context)));
 
                 AtlassianServerCapabilities capabilities = client.getCapabilityClient().getServerCapabilities();
-                if (credentials.filter(StringCredentials.class::isInstance).isPresent()) {
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java:378]: Why is this removed?

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java [hunk 2]
```diff
@@ -433,7 +433,7 @@ public FormValidation doCheckServerId(@AncestorInPath Item context,
 
         @Override
         @POST
-        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item context,
+        public HttpResponse doFillCredentialsIdItems(@AncestorInPath Item context,
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java:436]: ? `HttpResponse` why this change?

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java [hunk 3]
```diff
@@ -366,7 +366,7 @@ public FormValidation doCheckServerId(@AncestorInPath Item context,
 
         @Override
         @POST
-        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item context,
+        public HttpResponse doFillCredentialsIdItems(@AncestorInPath Item context,
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:369]: Again, same change, seems unrelated to the task at hand? I would have thought it would be better to stay as a `ListBoxModel` as that is what the FE needs. While this works, I don't understand why we've changed it?

I won't comment on all of them, but there are a few of these scattered across the changeset.

File: src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorIT.java [hunk 4]
```diff
@@ -114,38 +106,29 @@ public void testFindProjectsCredentialsIdInvalid() throws Exception {
...
     public void testFindProjectsCredentialsIdMissing() throws Exception {
-        BitbucketServerConfiguration bitbucketServerWithoutCredentials = new BitbucketServerConfiguration(
-                bbJenkinsRule.getBitbucketServerConfiguration().getAdminCredentialsId(),
-                bbJenkinsRule.getBitbucketServerConfiguration().getBaseUrl(), null, null);
-        bbJenkinsRule.addBitbucketServer(bitbucketServerWithoutCredentials);
-        HudsonResponse<List<JenkinsBitbucketProject>> response =
+        Response response =
                 RestAssured.expect()
-                        .statusCode(200)
+                        .statusCode(400)
```
Comment (dkjellin) [inline src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorIT.java:111]: so this is interesting, 
While I don't think we need to change this as part of this test, but what about public repositories? You don't need credentials for them. I would _assume_ that public repositories is not fantastically common among our users but now it feels like we're forcing you to use credentials when you actually don't need to.

**Review by dkjellin — APPROVED** [review]
Looks good, thank you for doing this. 
I think that we may need to increase our testing for public repositories maybe (but that is not in scope for this PR). Maybe you could do a manual test and see how that works before merging and we can decide if it is good enough or needs attention?

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java (hunk 1 above)
Comment (alibk0rd) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java:378]: Well, we no longer have `credentials` in the server config. Is there any other way to get the credentials for this check?

Comment (alibk0rd) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java:378]: P.S. why was this permission check limited to `StringCredentials` only?

File: src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorIT.java (hunk 4 above)
Comment (alibk0rd) [reply src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorIT.java:111]: Hmm, yes, I suppose that was always the case, except we're now more explicit about it?
I agree even if public repos aren't _common_ among our customers, we should still think about them. I'll dig a bit deeper. :+1:

Comment (alibk0rd) [reply src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorIT.java:111]: - [ ] make credentials optional for public repos?

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmFormValidationDelegate.java [hunk 5]
```diff
@@ -86,8 +97,7 @@ public FormValidation doCheckProjectName(@Nullable Item context, String serverId
                                 .getClient(
                                         serverConf.getBaseUrl(),
                                         jenkinsToBitbucketCredentials.toBitbucketCredentials(
-                                                providedCredentials.orElse(null),
-                                                serverConf.getGlobalCredentialsProvider("Check Project Name")));
+                                                providedCredentials.orElse(null)));
```
Comment (khughes-atlassian) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmFormValidationDelegate.java:100]: You know that `providedCredentials` is present, so its safe to call `.get`

**Review by khughes-atlassian — APPROVED** [review]

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java (hunk 1 above)
Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java:378]: This was checking if your token worked. `StringCredentials` are for bearer tokens, and that was how we used to check which of them failed, but you are probably right, it can go now, as we _know_ it is the PAT that's broken if we can't authenticate against the server.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmFormValidationDelegate.java (hunk 5 above)
Comment (alibk0rd) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmFormValidationDelegate.java:100]: - [ ] call `.get()` instead

**Review by dkjellin — APPROVED** [review]
